consensus/istanbul/backend/handler.go

```diff
@@ -72,6 +75,18 @@ func (sb *Backend) HandleMsg(addr common.Address, msg p2p.Msg, peer consensus.Pe
 			return true, errDecodeFailed
 		}
 
+		if msg.Code == istanbulDelegateSign {
+			if sb.config.Proxy {
```

Comment:
Seems like the same thing is being done, maybe condense this into `if (this is a proxy || this is proxied)` and return an error if neither